src/scripts/main.js

```diff
@@ -1396,6 +1397,45 @@
         }
     }
 
+    /**
+     * Report the tower stage - outside, eclipse, tower floor 1-7, 9-15, 25+-31+, umbra run
+     * @param {Object <string, any>} message The message to be sent.
+     * @param {Object <string, any>} user The user state object, when the hunt was invoked (pre-hunt).
+     * @param {Object <string, any>} user_post The user state object, after the hunt.
+     * @param {Object <string, any>} hunt The journal entry corresponding to the active hunt.
+     */
+    function addValourRiftStage(message, user, user_post, hunt) {
+        const attrs = user.environment_atts;
+        switch (attrs.state) {
+            case 'tower':
+                var floor, prefix;
+                if (attrs.floor % 8) {
+                    floor = 'Eclipse';
+                } else if (attrs.floor < 25) {
+                    floor = attrs.floor
+                } else {
+                    // Convert 25+ into 25-31
+                    floor = (((attrs.floor - 25) % 8) + 25) + '+'
+                }
+
+                if (attrs.active_augmentations.tu) {
+                    prefix = "UU Floor";
+                } else {
+                    prefix = "Floor";
+                }
+
+                message.stage = prefix + ' ' + floor
```

Comment:
I (and several other Discord users) think a plain `Eclipse` would be better than `Floor Eclipse`